src/main/java/org/testng/internal/Graph.java

```diff
@@ -121,7 +122,10 @@ public void topologicalSort() {
     // Sort the nodes alphabetically to make sure that methods of the same class
     // get run close to each other as much as possible
     //
-    Collections.sort(nodes2);
+    Comparator<Node> comparator = Systematiser.getComparator();
```

Comment:
From what I have looked at, if one were to be resorting to make this available via a configuration instead of a JVM argument, this is going to require quite a few changes. Also if you look at it, a configuration would still be applicable at the entire TestNG instance level, which is kind of what the JVM argument is doing as well. Does that add a bit of clarity to your question ?